File: public/sdk/node/examples/full-flow.js
const { ArianaEnterpriseClient } = require('../ariana-enterprise-sdk');

const api = new ArianaEnterpriseClient({ apiKey: process.env.ARIANA_API_KEY, environment: 'sandbox' });

async function main() {
  const sku = `ARI-SDK-${Date.now()}`;
  const orderId = `ARI-PED-SDK-${Date.now()}`;
  console.log(await api.health());
  console.log(await api.catalog.push([{ sku, name: 'Produto SDK Teste', price: 2299, stock: 10 }]));
  console.log(await api.products.updateStock(sku, { stock: 8 }));
  console.log(await api.products.updatePrice(sku, { price: 2199 }));
  console.log(await api.orders.create({ externalOrderId: orderId, customerName: 'Cliente SDK', customerEmail: 'cliente@teste.com', customerPhone: '31999999999', items: [{ sku, name: 'Produto SDK Teste', qty: 1, unitPrice: 2199, sellerId: 'ariana_moveis' }] }));
  console.log(await api.orders.invoice(orderId, { manufacturer: 'ariana_moveis', invoice: { number: '12345', series: '1', accessKey: '31260600000000000000550010000123451000012345', danfeUrl: 'https://teste.com/danfe.pdf', xmlUrl: 'https://teste.com/nfe.xml', total: 2199 } }));
  console.log(await api.orders.tracking(orderId, { manufacturer: 'ariana_moveis', trackingCode: 'TESTE123456789BR', carrier: 'Transportadora Ariana', trackingUrl: 'https://rastreamento.teste.com/TESTE123456789BR' }));
}
main().catch(err => { console.error(err); process.exit(1); });

File: public/sdk/python/ariana_enterprise_python/examples/full_flow.py
from ariana_enterprise import ArianaEnterpriseClient

api = ArianaEnterpriseClient(api_key="ari_sbx_xxxxx", environment="sandbox")

sku = "ARI-PY-0001"

print(api.health())
print(api.catalog.push([{"sku": sku, "name": "Produto Python", "price": 2299, "stock": 10}]))
print(api.products.update_stock(sku, 8))
print(api.products.update_price(sku, 2199))

order = api.orders.create({
    "manufacturer": "ariana_moveis",
    "externalOrderId": "PED-PY-0001",
    "customerName": "Cliente Python",
    "customerEmail": "cliente@teste.com",
    "customerPhone": "31999999999",
    "items": [{"sku": sku, "qty": 1, "unitPrice": 2199}]
})
print(order)

order_id = order.get("externalOrderId", "PED-PY-0001")
print(api.invoice.send(order_id, {"manufacturer": "ariana_moveis", "invoice": {"number": "12345", "series": "1", "danfeUrl": "https://teste.com/danfe.pdf", "xmlUrl": "https://teste.com/nfe.xml"}}))
print(api.tracking.update(order_id, {"manufacturer": "ariana_moveis", "trackingCode": "TESTE123BR", "carrier": "Transportadora Teste", "trackingUrl": "https://rastreamento.teste/TESTE123BR"}))
print(api.webhooks.test())
